### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by worldhck**
CMS module for creating menus

### CODE REVIEW

File: src/Http/Controllers/Admin/MenuController.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+        return $form;
+    }
+
+    /**
+     * @param Grid $grid
+     */
+    public function grid(Grid $grid)
+    {
+        $grid->column('name');
```
Comment (cliche23) [inline src/Http/Controllers/Admin/MenuController.php:46]: Maybe add the related node?

File: src/Menu/MenuFactory.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                if (!empty($link) && !empty($name)) {
+                    return [
+                        'link' => $link,
+                        'name' => $name,
+                        'active' => $this->isActive($link)
+                    ];
+                }
+                return false;
+            })
+            ->filter()
```
Comment (cliche23) [inline src/Menu/MenuFactory.php]: What does this do?

File: src/Menu/MenuFactory.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            ->toArray();
```
Comment (cliche23) [inline src/Menu/MenuFactory.php]: Is this needed? $items variable should already be a collection.

File: src/Menu/MenuFactory.php (hunk 2 above)
Comment (worldhck) [reply src/Menu/MenuFactory.php]: To process items the `map ` loop is used and if the item is has empty name and link then `false` will be returned instead to skip this item. 
By documentation: If no callback is supplied, all entries of the collection that are equivalent to false will be removed.

File: src/Http/Controllers/Admin/MenuController.php (hunk 1 above)
Comment (worldhck) [reply src/Http/Controllers/Admin/MenuController.php:46]: This is a god idea. At the moment menu module is not providing relation to node or specific object, it is done manually adding menu_id to node or specific objects. 
Do you mean finding node where this is related to or do you mean adding node relation to menu directly that can ease the implementation as well and there will not be a need to create a separate migrations?

File: src/Menu/MenuFactory.php (hunk 3 above)
Comment (worldhck) [reply src/Menu/MenuFactory.php]: This is not necessary. 
Fixed https://github.com/kristapsbworks/arbory-menu/commit/3487eeb40326f812c63c539ade9e59a2eb4dd7d1

File: src/Http/Controllers/Admin/MenuController.php (hunk 1 above)
Comment (cliche23) [reply src/Http/Controllers/Admin/MenuController.php:46]: Nevermind, I mixed up menu_items with menu. Closed.

**Pull request merged**